src/transformers/trainer.py

```diff
@@ -1096,7 +1096,35 @@ def call_model_init(self, trial=None):
 
         return model
 
-    def _wrap_model(self, model, training=True):
+    def torch_jit_model(self, model, training=False, dataloader=None):
+        if not training:
+            jit_inputs = ()
+            for _, batch in enumerate(dataloader):
+                for _, label in enumerate(batch):
```

Comment:
I'm not sure I understand what is happening here. The label name is not necessarily the last key in the batch, there is a `self.label_names` attribute for this.

Also why use `enumerate` if it's to discard the index?

Reply:
Here what we do is to construct the `jit_inputs `for `torch.jit.trace`, as steps:
(1) get the first batch from the dataloader as the `example_batch`
(2) enumerate each tensor from this `example_batch` and create each `example_tensor`
(3) tuple the `jit_inputs ` and it will be used in `torch.jit.trace`.

Therefore we do not need `self.label_names`, we are enumerating the datasets tensors ( batch: {key : tensor}, not related to labels)

Thanks.